### ISSUE TITLE
Privacy controls

### ISSUE THREAD

**Issue by ljoonal** [Neos-Metaverse/NeosPublic#3643]
### Is your feature request related to a problem? Please describe.

Remote users can access things like my local timezone and FPS. Also making me contact remote servers (with an image for example with a https:// url) is possible.

From those details a remote user can estimate somewhat accurately where I live and how good of a computer I have, as well as other details I'd consider to be my personal data. And they don't need any consent from me, and there is no indication that they have access to such data.

So the core of issue is, there is no requirement of consent for remote users to be able to get data which I consider to be personal data, nor is there any indication that it is happening.

### Relevant issues

Relevant documentation issue: #3405 
For remote resources being fetched, I couldn't find an exact issue about it, but there is some relevant documentation at https://wiki.neos.com/Guidelines#User_Data_Tracking

### Describe the solution you'd like

For a platform like Neos, I don't think that fully preventing personal data from being shared is possible. However, that doesn't mean that an effort shouldn't be made to help users who care about it control it more.

Designing systems with privacy in mind would be the best solution, but I imagine it's a bit too late for it as changing the way some of the LogiX nodes work would break a lot of things.

For the TimeZone and FPS examples, the simple solution would be to have an option to default them to some values such as UTC and 30 FPS. But at least for the FPS, it would most likely possible break some things, as to actually hide the it one would need to spoof the delta frame times as well.

For the remote resources being fetched, just optionally extending the current host access permissions seems like it would work rather well. This shouldn't break too too much, as such requests can already fail due to various reasons.

### Describe alternatives you've considered

- The more complex privacy solution for controlling the sharing of FPS and TimeZone and such values would be some sort of extended permissions system where accessing those values could require consent. 
  - While probably having the permission system default to allowing access though, as most users probably wouldn't care and would just find it annoying.
  - The few that do want more privacy could default it to deny permission and manually review what's trying to access those values.
- At least have an indication whenever something has/is accessing what could be considered personal data
  - It wouldn't solve the whole issue of not requiring consent
  - It'd at least let the user know that such data is being accessed
- Continuing on evaluating what GDPR actually legally requires and doing a bare minimum to comply with it.
  - Doesn't solve the problem described here, but completely legal and acceptable by most users

### Additional context

It was stated that mods should have references to issues here if they're solving problems.

**Comment by Lexevolution** [comment]
The team actually has a small section on what they consider private data on the wiki: https://wiki.neos.com/Security#Private_Data

**Comment by ljoonal** [comment]
Yes, which was already covered in the linked documentation issue and is more relevant to that I feel like than this feature request.

**Comment by Psychpsyo** [comment]
The issue with resources being fetched has been brought up here before in issue #883:

> For the remote resources being fetched, just optionally extending the current host access permissions seems like it would work rather well. This shouldn't break too too much, as such requests can already fail due to various reasons.

It would indeed probably not 'break' too much, but it would trigger a host access request on every single (external) resource. Someone spawned a video? Everyone gets a popup. Pasted an image? popup. Loaded an item that does so from their inventory? Potentially multiple popups for everyone at once.
From a security standpoint, that would be great. From a UX standpoint, that is abysmal.

**Comment by ljoonal** [comment]
> Someone spawned a video? Everyone gets a popup

Note that with the host domain checks, it doesn't re-ask after you've allowed/denied a domain once, so YouTube for example would only ask you if you've not already whitelisted or denied the domain(s) it uses before ever.

I understand though that it could be annoying for most users who don't care about privacy, which is why I suggested having it be optional. I can say though that it's not too bad after the initial batch of hosts, since people don't tend to have multiple personal domains, so the common and few personal domains are pretty quick to deny/allow.

**Comment by Enverex** [comment]
> > Someone spawned a video? Everyone gets a popup
> 
> Note that with the host domain checks, it doesn't re-ask after you've allowed/denied a domain once, so YouTube for example would only ask you if you've not already allowed or denied the domain(s) it uses before ever.
> 
> I understand though that it could be annoying for most users who don't care about privacy, which is why I suggested having it be optional. I can say though that it's not too bad after the initial batch of hosts, since people don't tend to have multiple personal domains, so the common and few personal domains are pretty quick to deny/allow.

The problem with this is that most people are idiots, or at least forgetful. I've had people complain so much about some of my worlds/tools not working after they refused the request which states it's required for it to work. Then they just assume the things are "broken" and not that they've explicitly broken them themselves by blocking access.
